upb/descriptor/reader.c

```diff
@@ -87,14 +90,20 @@ static char *upb_join(const char *base, const char *name) {
 }
 
 /* Qualify the defname for all defs starting with offset "start" with "str". */
-static void upb_descreader_qualify(upb_filedef *f, char *str, int32_t start) {
+static bool upb_descreader_qualify(upb_filedef *f, char *str, int32_t start) {
   size_t i;
   for (i = start; i < upb_filedef_defcount(f); i++) {
     upb_def *def = upb_filedef_mutabledef(f, i);
     char *name = upb_join(str, upb_def_fullname(def));
+    if (!name) {
+      /* Need better logic here; at this point we've qualified some names but
+       * not others. */
+      return false;
+    }
     upb_def_setfullname(def, name, NULL);
```

Comment:
There are probably lots of places like this, and we don't have a great recovery story here. I'm going to punt on being super thorough with this until we're actually testing for it anyway by injecting malloc failures.